settings/resource_default_namespace_settings.go

```diff
@@ -0,0 +1,162 @@
+package settings
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/settings"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// NewDefaultNamespaceSettingsAPI creates a DefaultNamespaceSettingsAPI instance
+func NewDefaultNamespaceSettingsAPI(ctx context.Context, m any) DefaultNamespaceSettingsAPI {
+	client := m.(*common.DatabricksClient)
+	return DefaultNamespaceSettingsAPI{client, ctx}
+}
+
+// DefaultNamespaceSettingsAPI exposes the Default Namespace Settings API
+type DefaultNamespaceSettingsAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+func (a DefaultNamespaceSettingsAPI) getCurrentEtag(etag string) (string, error) {
+	res, err := a.Read(etag)
+	if err != nil {
+		// If settings is not set-up, we will get an error with the etag in it.
+		// This will happen, for instance, during the first ever call to GET.
+		return a.getEtagFromError(err)
+	}
+	return res.Etag, nil
+}
+
+func (a DefaultNamespaceSettingsAPI) getEtagFromError(err error) (string, error) {
+	errorInfos := apierr.GetErrorInfo(err)
+	if len(errorInfos) > 0 {
+		metadata := errorInfos[0].Metadata
+		if etag, ok := metadata["etag"]; ok {
+			return etag, nil
+		}
+	}
+	return "", fmt.Errorf("error fetching the default workspace namespace settings: %w", err)
+}
+
+func (a DefaultNamespaceSettingsAPI) Create(request settings.UpdateDefaultWorkspaceNamespaceRequest) (string, error) {
+	return a.Update("", request)
```

Comment:
We don't expose a separate CREATE. Our UPDATE (patch) creates the resource if it does not exist. So the user will not get an error back if they call this CREATE method for a setting that already exists - it will be treated as yet another UPDATE. What is the expectation from TF perspective? By the way, access mgmt's CREATE is also that way - CREATE is treated as an update (https://github.com/databricks/terraform-provider-databricks/pull/2371/files#diff-7eef2f7ec147f91e0b6597efd71eeece37d9dbf900e83f03d8f5f00d6011b7e7R89-R99) @hectorcast-db

Reply:
As you can read in this code, the Create call just calls the Update one. I created a separate method because it feels more readable to me. But I can remove it.

Reply:
@mgyucht Is this behavior above acceptable for CREATE?

Reply:
I have done some changes, so this comment may not be applicable anymore. 
But what do you mean exactly?

Reply:
CREATE is based on UPDATE: https://github.com/databricks/terraform-provider-databricks/pull/2710/files#diff-d0b20ee3d8f9ca5a7dd83ed86b87a701fb91c302b9119d54945ecc13a40b696aR123. We do not have a POST RPC in our API, we only have PATCH and it does an UpSert (insert or update). What this means is that, the CREATE method on TF will not really be different from the UPDATE method and it will not fail if you call CREATE on an already existing setting (it will just update the existing entity). Is that the expected behavior? One can argue that CREATE should fail if the resource already exists.

I am looking at http://go/terraform-friendly and I don't see a note on CREATE to return ALREADY_EXISTS. In fact, the statement below hints that CREATE can "recreate" the resource instead of failing, if the resource already exists. So I believe we're fine here.

> An update could be omitted, but the resource is recreated with every change.